### ISSUE TITLE
Resident Evil Revelations 2 Glitches in game

### ISSUE THREAD

**Issue by sylkat** [Joshua-Ashton/d9vk#164]
The game is starting and after the CAPCOM->SIMPLY LOGO screen it crashes here is the backtrace:
```

Backtrace:
=>0 0xf76d8404 in libvulkan_radeon.so (+0x86404) (0xf19a85f0)
  1 0xf76d0b5d in libvulkan_radeon.so (+0x7eb5c) (0x0ed4e894)
  2 0xf76d44f6 in libvulkan_radeon.so (+0x824f5) (0x00000001)
  3 0xf76d64cc in libvulkan_radeon.so (+0x844cb) (0xf79cc1a0)
  4 0xf76d65a6 in libvulkan_radeon.so (+0x845a5) (0x0ed4f2a8)
  5 0x7e4bf0c1 wine_vkCreateGraphicsPipelines+0x1b0() in winevulkan (0x0ed4f208)
  6 0x6249aed1 in d3d9 (+0x5aed0) (0x0ed4fcb8)
  7 0x6249bef8 in d3d9 (+0x5bef7) (0x0ed4fd68)
0xf76d8404: movl	0xc(%eax),%esi
```
As i see in google maybe something related to an incompatible shaders..not sure really.

### Software information
Resident Evil Revelations 2, working well without d9vk and with gallium nine standalone.
Configured as windows 7 with all the dx9 dll to native.

### System information
- GPU: Amd Vega 11
- Driver: amdgpu
- Wine version: 4.9 staging
- DXVK version: 0.12/0.11

### Apitrace file(s)

http://www.mediafire.com/file/t111eq8a2k25ad1/ResidentEvilR2_d3d9.tar.xz/file

### Log files
[rerev2_d3d9.log](https://github.com/Joshua-Ashton/d9vk/files/3272281/rerev2_d3d9.log)

**Comment by neVERberleRfellerER** [comment]
Looks like driver bug. The same as here
https://github.com/Joshua-Ashton/d9vk/issues/64#issuecomment-491299122
explained here
https://github.com/Joshua-Ashton/d9vk/issues/64#issuecomment-491488934
and fixed in mesa-git.
Your mesa is also greatly outdated.

**Comment by sylkat** [comment]
Yep as i see in those comments they got fixed the problem with mesa 19.1 i have 18.3.6 im in debian testing.
Maybe i will upgrade mesa but sometimes i got broken system upgrading with packages i have to compile myself probably.

**Comment by sylkat** [comment]
Confirmed, with the last mesa 19.2 i dont have crash anymore.

But there is some strange behaviour in the graphics, the character is not displayed well and some objects are not shown correctly, i dont know about 3d so i cant explain correctly whats going on.
I attach a picture of the issue.
 

![residentdxvk](https://user-images.githubusercontent.com/4473147/59214900-9c283680-8bea-11e9-9575-031161bbb948.png)

**Comment by neVERberleRfellerER** [comment]
For graphical glitches, capture apitrace as decribed here
https://tracehub.froggi.es/
Do this with native d3dx9 DLLs. I guess you can upload the apitrace where you want, not just to tracehub if you don't want to.

**Comment by sylkat** [comment]
Here is the apitrace:

http://www.mediafire.com/file/t111eq8a2k25ad1/ResidentEvilR2_d3d9.tar.xz/file

I will update the first post to add it there as well.

And some screenshots of the black artifacts shown in the game:

![resident1](https://user-images.githubusercontent.com/4473147/59251042-6e2f0a80-8c5b-11e9-83f0-22ee0c586213.png)
![resident2](https://user-images.githubusercontent.com/4473147/59251043-6e2f0a80-8c5b-11e9-9ac3-c880864f4a9f.png)
![resident3](https://user-images.githubusercontent.com/4473147/59251044-6ec7a100-8c5b-11e9-9b80-fffa5cbed89b.png)

**Comment by NRngnl** [comment]
I don't think this issue is related to #99, if #99 is applied to AMD/Intel only . 
I am using Nvidia Geforce RTX 2070, but I have the same glitches as the screenshot provided above.
I found the same glitch not only appears in Resident Evil Revelations 2 only, but also appears in Resident Evil 6. You can see the 6 logo repeated several times on top left region of the main menu.

![Resident Evil 6   Biohazard 6 Screenshot 2019 06 17 - 21 19 33 27](https://user-images.githubusercontent.com/27538249/59607392-ddb66580-9145-11e9-90b2-e8af146b4f45.png)


![Resident Evil 6   Biohazard 6 Screenshot 2019 06 17 - 21 19 33 27](https://user-images.githubusercontent.com/27538249/59607607-3ede3900-9146-11e9-9abc-43ce9b357233.png)

**Comment by NerosTie** [comment]
![Screenshot_20190925_134737](https://user-images.githubusercontent.com/4572452/65598188-27131e00-df9b-11e9-8458-545558bbb2f4.png)


I see no glitches at all. Maybe the issue can be closed? (tested with 0.21)

**Comment by Joshua-Ashton** [comment]
Looks to work fine now in AMD and NV in the trace you sent. Feel free to re-open if you have any more issues.
